### ISSUE TITLE
Sometimes module autoloading loads VMware module instead of Hyper-V module

### ISSUE THREAD

**Issue by csandfeld** [VirtualEngine/Lability#333]
Every now and then, I end up in a situation where VMware’s PowerCLI module is autoloaded instead of Microsoft’s Hyper-V module, resulting in errors when running Lability commands.

I think the best fix for this problem would be to use module-qualified cmdlet names when calling cmdlets from the Hyper-V module. If you agree, I will be happy to take on the task of fixing that and submit a PR.

Thoughts?

**Comment by ChrisLGardner** [comment]
The other option is to update the module manifest and add Hyper-v to required modules. Using fully qualified command names is definitely a good step as well for general best practices.
________________________________

**Comment by csandfeld** [comment]
Using the module-qualified names will also help in cases where the VMware modules are already loaded before loading the Lability module.

I thought about adding to required modules in the manifest as well, but wasn’t sure if that could cause other problems, such as not being able to import the Lability module if the Hyper-V module is not present. Today (if I remember correctly) we can use Start-LabHostConfiguration to setup the host, including installation of the Hyper-V role and module. Would it still be possible to install and import the Lability module on a host where the Hyper-V module is not yet present?

What's you take on this @iainbrighton ?

**Comment by iainbrighton** [comment]
Thinking out loud, wouldn't using fully-qualifying cmdlets cause an issue with the unit tests? Could we not explicitly import the Hyper-V module in the `Lability.psm1` file? From within the module's scope, it *should* use these cmdlets - regardless of what is loaded in the global/calling scope?

**Comment by ChrisLGardner** [comment]
Pester will correctly handle fully qualified names and still call the relevant mocks.

**Comment by iainbrighton** [comment]
@ChrisLGardner @csandfeld If you're both happy that module-qualified names is the way to go (and won't impact the testing) then I say "Go for it!" 😃.

**Comment by csandfeld** [comment]
I already made the change in a branch of my fork, for both the Hyper-V and the Storage commands. Pester handles it just fine, except for some errors when invoking the tests in 
[Expand-LabImage.Tests.ps1](https://github.com/VirtualEngine/Lability/blob/76a79cf20c469343921a32e053a9a859b3042dd9/Tests/Unit/Src/Private/Expand-LabImage.Tests.ps1).

```PowerShell
Cannot process argument transformation on parameter 'DiskImage'. Cannot convert the "@{DriveLetter=Z}" value of type "System.Management.Automation.PSCustomObject" to type "Microsoft.Management.Infrastructure.CimInstance
```

I can see that the problem was suppressed before by "overwriting" the Get-Volume function in the tests file ...

https://github.com/VirtualEngine/Lability/blob/76a79cf20c469343921a32e053a9a859b3042dd9/Tests/Unit/Src/Private/Expand-LabImage.Tests.ps1#L12-L22

... but that workaround no longer work when using a module-qualified call to Get-Volume in [Expand-LabImage.ps1](https://github.com/VirtualEngine/Lability/blob/76a79cf20c469343921a32e053a9a859b3042dd9/Src/Private/Expand-LabImage.ps1). Not sure how to get around that problem yet.

**Comment by csandfeld** [comment]
@iainbrighton explicitly loading the module in Labilit.psm1 might work as well. We would just need to consider that the Hyper-V module may not be installed on the box when we import the module (since we can use Start-LabHostConfiguration to install that).

**Comment by csandfeld** [comment]
I fixed the failing test by mocking Get-DiskImage with a deserialized MSFT_DiskImage CimInstance object.
I will submit a PR shortly.

**Comment by iainbrighton** [comment]
Closed in #334
